### PR TITLE
#747 automatic frontend translations

### PR DESCRIPTION
**Pull request by rizkypujiraharja**
Command `php artisan chatgpt:translate-app`

The concept is that we only need to write our text in the en.json file. Then, the program will read the list of directories in the locales folder. After that, the other language files and the en.json file will be merged.

Once the files are merged, ChatGPT will only translate the entries whose values are empty. After receiving the translations from ChatGPT, each file will be rewritten according to its language code.

Command `php artisan app:generate-translation-text {path : directory or file}`

After we update the text for translation in the Vue file, run the command above to extract all the text for translation and add it to the en.json file.

### CODE REVIEW

**Review by ArturHanusek — CHANGES_REQUESTED** [review]
We need to write the test for it. 

When writing tests for feature which requires env variable to be set:
- first check if the env variable is set - mark test skipped if not ()
- perform test action

**Comment by rizkypujiraharja** [comment]
https://jam.dev/c/624175e5-0da7-48e0-8fa4-33aac56b8b99

File: app/Console/Commands/ChatgptTranslateApp.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+     * Execute the console command.
+     */
+    public function handle()
+    {
+        $dirPath = base_path('resources/js/locales');
+        $enFileJson = file_get_contents($dirPath . '/en.json');
+        $enFileCollecton = collect(json_decode($enFileJson, true));
+
+
+        $listFiles = $this->getFiles($dirPath);
```
[the remark this answers is not in the document]
Comment (rizkypujiraharja) [reply app/Console/Commands/ChatgptTranslateApp.php:41]: Noted

File: app/Console/Commands/ChatgptTranslateApp.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+     */
+    protected $description = 'Generate localization based on en.json';
+
+    /**
+     * Execute the console command.
+     */
+    public function handle()
+    {
+        $dirPath = base_path('resources/js/locales');
+        $enFileJson = file_get_contents($dirPath . '/en.json');
```
[the remark this answers is not in the document]
Comment (ArturHanusek) [reply app/Console/Commands/ChatgptTranslateApp.php]: i think its good idea!

**Review by ArturHanusek — APPROVED** [review]

**Pull request merged**